### PR TITLE
Fix up transformer logging

### LINKED ISSUE #3051: transformer: do not share a global logger among transformers
Currently most transformers are importing and using a single logger from `package:angular2/src/transform/common/logging.dart`. Unfortunatelly, transformers may be interleaved at any point where async calls are made, so as a result pub might report errors in the wrong file.

We should properly pass the logger through all calls made by a transformer.

@jakemac53 @kegluneq

### PR DESCRIPTION
**Pull request by jakemac53**
fixes https://github.com/angular/angular/issues/3051

### CODE REVIEW

File: modules/angular2/src/reflection/debug_reflection_capabilities.dart [hunk 1]
```diff
@@ -12,8 +12,9 @@ class ReflectionCapabilities extends standard.ReflectionCapabilities {
   ReflectionCapabilities({bool verbose: false})
       : _verbose = verbose,
         super() {
-    Logger.root.level = _verbose ? Level.ALL : Level.INFO;
-    Logger.root.onRecord.listen((LogRecord rec) {
+    hierarchicalLoggingEnabled = true;
```
Comment (kegluneq) [inline modules/angular2/src/reflection/debug_reflection_capabilities.dart]: Where is this defined? Perhaps it should be imported via prefix to make it clear?

**Comment by kegluneq** [comment]
So... after looking at this for a bit I had an idea: what do you think about using [zone local values](https://www.dartlang.org/articles/zones/#storing-zone-local-values) instead of plumbing `logger` around to all the code?

I think using zones would limit the changes to:
```
// logging.dart
Future init(Transform t) {
  var completer = new Completer();
  runZoned(() {
    completer.complete(null);
  }, zoneValues: { #logKey: new BuildLogger(t) });
  return completer.future;
}

BuildLogger get logger {
  var logger = Zone.current[#logKey] as BuildLogger;
  return logger != null ? logger : new PrintLogger();
}

// .../*/transformer.dart
Future apply(Transform transform) async {
  log.init(transform).then((_) {
    _actualTransformWork();
  });
}
```

Sorry for not thinking of this before, but what do you think?

**Comment by jakemac53** [comment]
I didn't even know zone local values existed haha, but looks like a good solution to me! I will try that out.

**Comment by jakemac53** [comment]
ok, this is now updated per offline discussion

File: modules/angular2/src/transform/common/logging.dart [hunk 2]
```diff
@@ -5,24 +5,27 @@ import 'package:barback/barback.dart';
...
-/// The logger the transformer should use for messaging.
+dynamic setZoned(BuildLogger logger, _SimpleCallback fn,
+    {String errorMessage: ''}) => runZoned(fn,
+        zoneValues: {_key: logger}, onError: (e, stackTrace) {
+  logger.error('$errorMessage\n'
+      'Exception: $e\n'
+      'Stack Trace: $stackTrace');
+});
+
+/// The logger for the current {@link Zone}..
```
Comment (kegluneq) [inline modules/angular2/src/transform/common/logging.dart]: nit: Ends with double period.

Comment (jakemac53) [inline modules/angular2/src/transform/common/logging.dart]: done

**Comment by kegluneq** [comment]
LGTM if LGTTravis

**Pull request merged**